django/db/migrations/state.py

```diff
@@ -180,10 +179,10 @@ def __init__(self, real_apps, models, ignore_swappable=False):
                 # (only for the first issue). Error message should look like:
                 # "ValueError: Lookup failed for model referenced by
                 # field migrations.Book.author: migrations.Author"
-                msg = "Lookup failed for model referenced by field {field}: {model[0]}.{model[1]}"
-                raise ValueError(msg.format(field=operations[0][1], model=lookup_model))
+                msg = "Lookup failed for model: {model[0]}.{model[1]}"
+                raise ValueError(msg.format(model=lookup_model))
             else:
-                do_pending_lookups(model)
+                self.do_pending_operations(model)
```

Comment:
On my pull request #4097 I don't see the `else` executed either in Django's own test suite. The `except`, however, and the `raise ValueError()` inside is actively used. The entire for loop can practically be replaced by the same code which is used during Django's boot process: "here are a bunch of Python modules, go and find all model classes in there and register all relations".